src/bin/pg_autoctl/cli_show.c

```diff
@@ -400,12 +401,121 @@ cli_show_state(int argc, char **argv)
 }
 
 
+/*
+ * cli_show_local_state implements pg_autoctl show state --local, which
+ * composes the state from what we have in the configuration file and the state
+ * file for the local (keeper) node.
+ */
+static void
+cli_show_local_state()
+{
+	KeeperConfig config = keeperOptions;
+
+	switch (ProbeConfigurationFileRole(config.pathnames.config))
+	{
+		case PG_AUTOCTL_ROLE_MONITOR:
+		{
+			log_error("pg_autoctl show state --local is not supported "
+					  "on a monitor");
+			exit(EXIT_CODE_MONITOR);
+		}
+
+		case PG_AUTOCTL_ROLE_KEEPER:
+		{
+			bool missingPgdataIsOk = true;
+			bool pgIsNotRunningIsOk = true;
+			bool monitorDisabledIsOk = true;
+
+			Keeper keeper = { 0 };
+			CurrentNodeState nodeState = { 0 };
+
+			if (!keeper_config_read_file(&config,
+										 missingPgdataIsOk,
+										 pgIsNotRunningIsOk,
+										 monitorDisabledIsOk))
+			{
+				/* errors have already been logged */
+				exit(EXIT_CODE_BAD_CONFIG);
+			}
+
+			if (!keeper_init(&keeper, &config))
+			{
+				/* errors have already been logged */
+				exit(EXIT_CODE_BAD_CONFIG);
+			}
+
+			/* also grab the minimum recovery LSN if that's possible */
+			if (!pg_controldata(&(config.pgSetup), missingPgdataIsOk))
```

Comment:
I'm not sure which of the available LSN locations we should display here. That said you're right, the current one that I've chosen is confusing for the user. I'm going to switch to using the "Last checkpoint location" which seems less confusing from a quick local test:

```
$ pg_autoctl show state
Name |  Node |      Host:Port |     Current State |    Assigned State |               LSN | Health
-----+-------+----------------+-------------------+-------------------+-------------------+-------
   a |     1 | localhost:5001 |           primary |           primary |         0/5000640 |    ✓
   b |     2 | localhost:5002 |         secondary |         secondary |         0/5000640 |    ✓
   c |     3 | localhost:5003 |         secondary |         secondary |         0/5000640 |    ✓

$ pg_controldata /tmp/pgaf/c | grep location
Latest checkpoint location:           0/4000060
Latest checkpoint's REDO location:    0/4000028
Minimum recovery ending location:     0/4000100
Backup start location:                0/0
Backup end location:                  0/0
```

Reply:
maybe if postgres is up we can ask it using one of the UDFs. If it's down, I guess it would be one of the latest checkpoint location reltaed ones.

Reply:
My understanding is that should we restart the secondary we will see the LSN get in the past and forward again after crash recovery is done.  We might want to change the column name when using `--local`, also.

Reply:
We're now using the LSN obtained connecting to Postgres when it's running, so that's the same LSN that we send to the monitor. When Postgres is not running we grab the Latest checkpoint location from the output of pg_controldata.